### ISSUE TITLE
Can I use `--gen-spec-lib` to generate intermediate `report-lib.tsv` similar to quant files are merge later?

### ISSUE THREAD

**Issue by hguturu** [vdemichev/DiaNN#760]
DIA-NN has the nicely solves the N+1 problem by generating `.quant` files that can be quickly reprocessed when a new file is added and generates a full study `report.tsv`.

Is there a way to similarly regenerate the DIA library this way? I noticed I can use `--gen-spec-lib` to generate a `report-lib.tsv` per file.

I then wrote a simple script that goes through the individual reports and grabs the "best" spectra based on the `Q.Value` from the full study `report.tsv` filtering for precursors `Global.Q.Value`. This gets me a final library is very similar to the `report-lib.tsv` that gets generated if I run `--gen-spec-lib` on the final group run, but it is slightly different (e.g., some fragment ions are different and the fragment intensities are different).

Would this be an acceptable way to build the DIA library if there are lots of files to reprocess? Any suggestions on how to better merge the individial `report-lib.tsv`? Or can DIA-NN do this?

**Comment by vdemichev** [comment]
Yes, you can generate a lib like this :) You can also pull RT alignment from the main report. In most cases, keeping predicted spectra is fine (IDs, RTs and IMs profiling mode). But why do this manually if DIA-NN does this automatically?

**Comment by hguturu** [comment]
Ah got it, using the `iRT` from the report to replace `Tr_recalibrated` in the library to improve performance. I can try that.

I also compared keeping the subset of predicted spectra (e.g., using --gen-spec-lib with .quant files, but empty .mzML) files to run the final grouprun. This merged library is slightly better than just sub-setting the predicted spectra probably due to some experimental refining at individual run level.

*As to why?* We keep most of our raw data on s3 and only pull for search. For large groupruns, we are able to run with just the .quant files since the raw files arent really needed. But for library generation this requires downloading all the raw data and we just dont have drives large enough. Also its very slow to download all raw files. If DIA-NN could do this from the intermediate library files similar to how it does with intermediate .quant files, that would be awesome!

**Comment by vdemichev** [comment]
Then I would suggest to just use 'IDs, RT and IMs' profiling for library generation, which works with .quant files only. Raw files are needed to extract spectra, but in general predicted spectra are usually equally good as empirical DIA-derived ones.

**Comment by hguturu** [comment]
Is that the same library that gets generated at the final group run if you use --gen-spec-lib or --reanalyse without the raw files? A library file gets generated with warnings of unable to open file and then it obviously crashes when it tries to reanalyse due to not having the raw files.

**Comment by vdemichev** [comment]
>  A library file gets generated with warnings of unable to open raw files
Will not happen in IDs, RTs and IMs profiling. Just need to copy the command used to run diann.exe in this mode.

**Comment by hguturu** [comment]
I noticed `--gen-spec-lib` filters the `report-lib.tsv` to only precursors that meet `Global.Q.Value < 0.01`. Is there a way to change this to get more precursors in the library?
